session.go

```diff
@@ -4971,9 +5137,20 @@ func (c *Collection) writeOp(op interface{}, ordered bool) (lerr *LastError, err
 	s.m.RLock()
 	safeOp := s.safeOp
 	bypassValidation := s.bypassValidation
+	enableOpMsg := s.experimental["opmsg"]
 	s.m.RUnlock()
 
-	if socket.ServerInfo().MaxWireVersion >= 2 {
+	serverInfo := socket.ServerInfo()
+
+	if enableOpMsg && serverInfo.MaxWireVersion >= 6 {
+		// we can use OP_MSG introduced in Mongodb 3.6
+		oPlerr, oPerr := c.writeOpWithOpMsg(socket, serverInfo, op, ordered, bypassValidation, safeOp)
+		if oPlerr != nil || oPerr != nil {
```

Comment:
I think you can just drop the conditional and do:

```go
return c.writeOpWithOpMsg(socket, serverInfo, op, ordered, bypassValidation, safeOp)
```

Reply:
can't do this here, because some command in MongoDB 3.6 still use the old wire protocol. Without the conditional, the method would sometimes return before actually sending the command to the database